Cody completions: Add docs page (#52447)
sourcegraph/sourcegraph: doc/cody/completions.md

> # Cody completions

Reviewer: should we say anything about that the admin needs to configure a completionsModel in site-config?

Reviewer: and that only claude / llmproxy w/ claude work with that right now, but not GPT

Author: @eseliger How does the updated text look like for you? I’m not super familar with the setup so if you have any suggestions, please let me know :D

<aside class="experimental">
<p>
<span class="badge badge-experimental">Experimental</span> This feature is experimental and might change or be removed in the future. We've released it as an experimental feature to provide a preview of functionality we're working on.
</p>
</aside>

![Example of Cody completions. You see a code snippet starting with async function getWeather(city: string) { and Cody response with a multi-line suggestion using a public weather API to return the current weather ](https://storage.googleapis.com/sourcegraph-assets/cody_completions.png)

## What are Cody completions

 Cody Completions provide suggestions for how to continue your code at the current cursor position.

## Enabling Cody completions

While in experimental state, Cody completions need to be enabled manually. To do that:

1. Make sure your Cody AI by Sourcegraph extension is on the latest version

>    - <kbd>shift</kbd>+<kbd>cmd</kbd>+<kbd>x</kbd> to see all extensions, select Cody, confirm the version

Reviewer 2: How about
```suggestion
   - <kbd>Control + Shift + X</kbd> (<kbd>⌘⇧X</kbd>) to see all extensions, select Cody, confirm the version
```

# Cody completions

<aside class="experimental">
<p>
<span class="badge badge-experimental">Experimental</span> This feature is experimental and might change or be removed in the future. We've released it as an experimental feature to provide a preview of functionality we're working on.
</p>
</aside>

![Example of Cody completions. You see a code snippet starting with async function getWeather(city: string) { and Cody response with a multi-line suggestion using a public weather API to return the current weather ](https://storage.googleapis.com/sourcegraph-assets/cody_completions.png)

> ## What are Cody completions

Reviewer 3: ```suggestion
## What are Cody completions?
```

> Cody Completions provide suggestions for how to continue your code at the current cursor position while you're coding.

Author: @iskyOS @gwestersf @mrnugget I could use some help with the marketing language to use here 🙂

Reviewer 4: @philipp-spiess How's this? 

```suggestion
Cody provides real-time code completions as you're typing. As you start coding, or after you type a comment, Cody will look at the context around your cursor to predict what you're trying to implement and provide completions.```

> Cody provides real-time code completions as you're typing. As you start coding, or after you type a comment, Cody will look at the context around your open files and file history to predict what you're trying to implement and provide completions.

Reviewer 3: Maybe that's just me, but I think most discussions about "completions" become clearer once the word "autocomplete" is used :)

```suggestion
Cody provides real-time code completions as you're typing. As you start coding, or after you type a comment, Cody will look at the context around your open files and file history to predict what you're trying to implement and provide completions. It's autocomplete powered by Cody!
```

## Enabling Cody completions

While in experimental state, Cody completions need to be enabled manually. To do that:

> 1. Make sure your Cody AI by Sourcegraph extension is on the latest version

Reviewer 3: Link to extension in VS Code marketplace?

   - <kbd>shift</kbd>+<kbd>cmd</kbd>+<kbd>x</kbd> to see all extensions, select Cody, confirm the version

> 1. Next, go to the Cody Extension Settings and enable completions

Reviewer 3: ```suggestion
1. Go to the Cody Extension Settings and enable completions
```

In a numbered list of instructions I don't think we need to add "Next" :)

   - Click to check the box for: `Cody > Experimental Suggestions`

> 1. Finally, restart VS Code and test it out!

Reviewer 3: Doesn't reloading work too?

Author: Yes it does! I'll change the wording (and hopefully we can make reloads unnecessary too)

### Configuring on Sourcegraph Enterprise

Please follow the steps in [Enabling Cody on Sourcegraph Enterprise](.explanations/enabling_cody_enterprise#enabling-cody-on-sourcegraph-enterprise) to enable Cody on Sourcegraph enterprise.

> You can configure the model used for Cody completions via the `completionModel` option inside the `completions` site config.

Reviewer: ```suggestion
You have to configure the model used for Cody completions via the `completionModel` option inside the `completions` site config.
```

not an optional step :)

Author: Haha I did not read that this is temporary :P 

<img width="722" alt="Screenshot 2023-05-25 at 17 37 40" src="https://github.com/sourcegraph/sourcegraph/assets/458591/c7b89bbe-ddd8-4668-890a-01b4248f848e">

> Please follow the steps in [Enabling Cody on Sourcegraph Enterprise](.explanations/enabling_cody_enterprise#enabling-cody-on-sourcegraph-enterprise) to enable Cody on Sourcegraph enterprise.

Reviewer 3: ```suggestion
Please follow the steps in [Enabling Cody on Sourcegraph Enterprise](.explanations/enabling_cody_enterprise#enabling-cody-on-sourcegraph-enterprise) to enable Cody on Sourcegraph Enterprise.
```

> NOTE: Self-hosted customers need to update to a minimum of version 5.0.4 to use completions.

> > NOTE: Cody completions currently only work with Claude Instant or our LLMProxy configured with Claude Instant. Support for other models will be coming later.

Reviewer 3: I'd leave the "You have to configure the model" sentence off then

Author: > I'd leave the "You have to configure the model" sentence off then

Not sure I understand this suggestion: I think there's still value in telling users they have to set the config even though as a note they can only use claude yet. Might make sense to include the exact string though, will add that.

Author: I've done that now, wdyt about the latest state?
